makefile

```diff
@@ -0,0 +1,11 @@
+all: generate qkchashmake
+
+generate:
```

Comment:
1. I think naming it as `proto` is more explicit
1. can you give the prerequisites to be the proto file? which means if it changes, `make` can re-run the rule https://www.gnu.org/software/make/manual/make.html#Rule-Introduction

Reply:
Done.